services/src/main/java/org/keycloak/services/resources/account/AccountRestService.java

```diff
@@ -469,6 +456,65 @@ public LinkedAccountsResource linkedAccounts() {
         return new LinkedAccountsResource(session, request, client, auth, event, user);
     }
 
+    @Path("/applications")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @NoCache
+    public Response applications() {
+        checkAccountApiEnabled();
+        auth.requireOneOf(AccountRoles.VIEW_APPLICATIONS, AccountRoles.VIEW_PROFILE);
+
+        List<String> inUseClients = getInUseClients();
+
+        List<ClientRepresentation> apps = new LinkedList<>();
+        for (ClientModel client : realm.getClients()) {
+            if (client.isBearerOnly() || client.getBaseUrl() == null) {
+                continue;
+            }
+            apps.add(modelToRepresentation(client, false, false, inUseClients));
+        }
+
+        for (ClientModel client : new UserSessionManager(session).findClientsWithOfflineToken(realm, user)) {
+            apps.add(modelToRepresentation(client, false, true, inUseClients));
+        }
+
+        for (UserConsentModel consent : session.users().getConsents(realm, user.getId())) {
+            ClientModel client = consent.getClient();
+
+            if (!new StorageId(client.getId()).isLocal()) {
+                apps.add(modelToRepresentation(client, true, false, inUseClients));
+            }
+        }
+
+        return Cors.add(request, Response.ok(apps)).auth().allowedOrigins(auth.getToken()).build();
+    }
+
+    private List<String> getInUseClients() {
+        List<String> inUseClients = new LinkedList<>();
+        List<UserSessionModel> sessions = session.sessions().getUserSessions(realm, user);
+        sessions.addAll(session.sessions().getOfflineUserSessions(realm, user));
```

Comment:
Should we really mark offline sessions as "in-use"?

Reply:
@vmuzikar This was requested in the last round of reviews.

Reply:
Right, sorry.